### DISCUSSION TITLE
官网 "在线演示" 源码开源了吗？

### DISCUSSION THREAD

**Discussion by lsg1228** [jd-opensource/micro-app#173] [Show and tell]
官网 "在线演示" 源码有吗

**Comment by liuxiaoxiao1** [comment]
同求，想看下同一个页面里多个子应用是怎么配置的

**Comment by bailicangdu** [comment]
地址 https://github.com/micro-zoe/micro-app/tree/dev/dev

**Reply by Kilims** [reply]
若是ng配置也有，那再好不过了[笑脸] @bailicangdu 

拿着练手项目玩了下，能倒是能跑，有问题。

描述如下：

这是打算嵌入的子项目地址：https://yingtai.tech/cars_saling/MerchantPage
这是打算做基座的父项目地址：https://yingtai.tech/escape_online_booking/

一进到父项目地址，倒是能正常看到微前端的效果，但是url会直接跳转到”https://yingtai.tech/cars_saling/MerchantPage“。
使用上倒是没啥毛病，但如果当前页面refresh，就会直接单独打开子项目了。

怀疑我的nginx不适配，想参考下例子的ng配置。

感谢大佬，有回必三连

**Reply by bailicangdu** [reply]
子应用的baserouter设置了吗

**Reply by Kilims** [reply]
1. 子应用baserouter
2. 父应用路由
3. 父应用调用micro-app设置路由

按照[官网快速开始](https://cangdu.org/micro-app/docs.html#/zh-cn/start)配了的，详情如下


子应用
```
let routeName = "cars_saling"
<BrowserRouter basename={window.__MICRO_APP_BASE_ROUTE__ || `/${routeName}`}>
...
</BrowserRouter
```

父应用里

引用micro-app的Page组件
```
<micro-app
  name="cars_saling"
  url="https://yingtai.tech/cars_saling/"
  baseroute="/cars_saling"
></micro-app>
```

父应用的路由配置
```
let routeName = "escape_online_booking"
<BrowserRouter basename={`/${routeName}`}>
         <Route path="/cars_saling">
            <CarsSalingPage />
          </Route>
</BrowserRouter
```

**Reply by bailicangdu** [reply]
这里的值应该是 '/escape_online_booking/cars_saling'
![image](https://user-images.githubusercontent.com/20297227/145748766-0391a24b-cbee-4ca1-aaae-140474713606.png)

**Reply by Kilims** [reply]
花了点时间研究了下，好像悟了

单独打开子应用：https://yingtai.tech/sub-react/
父应用：https://yingtai.tech/admin/sub-react/

```
这里的值应该是 '/escape_online_booking/cars_saling'
```
哈哈对的，我也才发现了，改了之后就好了

接踵这玩意，没发现啥影响？

![image](https://user-images.githubusercontent.com/22916814/145749039-f87d7ea7-fb45-44cf-819f-bdbcb2a087fb.png)

**Reply by bailicangdu** [reply]
这个报错参考 https://micro-zoe.github.io/micro-app/docs.html#/zh-cn/framework/react?id=%e5%b8%b8%e8%a7%81%e9%97%ae%e9%a2%98

**Reply by Kilims** [reply]
舒服了，loop官网的时候瞄到过但没回想起来，灰常感谢！

**Reply by 333-zy** [reply]
https://github.com/micro-zoe/micro-app/tree/dev/dev这个地址失效了

**Comment by elarbor** [comment]
[官方示例项目](https://github.com/jd-opensource/micro-app-demo)好像是[v0](https://jd-opensource.github.io/micro-app/0.x/docs.html#/)的。我有个简化版的v0项目[micro-app-demo.zip](https://github.com/user-attachments/files/19676433/micro-app-demo.zip) 主：vue3 ； 子：vue2
